adal/src/main/java/com/microsoft/aad/adal/BrokerAccountServiceHandler.java

```diff
@@ -324,13 +330,41 @@ private void bindToBrokerAccountService(final Context context, final Callback<Br
         if (brokerEvent != null) {
             brokerEvent.setBrokerAccountServiceBindingSucceed(serviceBound);
         }
+        if (!serviceBound) {
+            connection.unBindService(context);
+            callback.onError(new AuthenticationException(ADALError.BROKER_BIND_SERVICE_FAILED));
+        } else {
+            // make sure it's bound to the service with success in a period not longer than the TIMEOUT
+            mUiHandler.postDelayed(timeoutRunnable, BIND_SERVICE_TIMEOUT_THRESHOLD);
```

Comment:
What kind of exception will be thrown if the service bounding reaches the time threshold? Will it return a different errorCode compared to ADALError.BROKER_BIND_SERVICE_FAILED?

Reply:
it will throw an AuthenticationException of "ADALError.BROKER_BIND_SERVICE_FAILED". "ADALError.BROKER_BIND_SERVICE_FAILED" is a newly added error. please check line 357. thanks

Reply:
Thanks for the reply. Is it possible to add the cause details in to the AuthenticationException(ADALError code, String details)? Currently, there is two cases here, Line 335 and Line 357. And it is difficult to tell the difference by ADALError only. Please correct me if there is anything I misunderstood. Thanks.

Reply:
good point, I've made the change. thanks